### PR TITLE
LayerScale fix to support pretrained ConvNeXt models with mixed precision

### PR DESCRIPTION
**Pull request by andreped**
Fix for the issue https://github.com/keras-team/keras/issues/16907. The proposed solution was proposed by @zibbini, and we both tested and verified that it works.

This PR fixes a bug in the [LayerScale](https://github.com/keras-team/keras/blob/master/keras/applications/convnext.py#L199) layer, which hindered ConvNeXt pretrained models to work with mixed precision. The solution was simply to set the appropriate dtype in the `gamma` variable of the LayerScale layer.

See [original gist](https://colab.research.google.com/gist/tilakrayal/30257452eebf9cc45e590f84ed8da4bd/untitled503.ipynb) to reproduce the bug.

See [new gist](https://colab.research.google.com/gist/andreped/5bc4393b64dbfcfed4c87e5c6863f3df/layerscale-mixed-precision-fix.ipynb) for verifying that the proposed fix works.

### CODE REVIEW

**Comment by Frightera** [comment]
You'll have an issue with formatting as there is no `,` at the end:

You have:

```
tf.Variable(
            self.init_values * tf.ones((self.projection_dim,)),
            dtype=self._compute_dtype_object
        )
```

Should have been:

```
tf.Variable(
            self.init_values * tf.ones((self.projection_dim,)),
            dtype=self._compute_dtype_object,
        )
```

Please check the [gist here](https://colab.research.google.com/gist/Frightera/8147f5d4d22b60025b784801da65a47c/17547-lint.ipynb). Thanks!

**Comment by andreped** [comment]
> You'll have an issue with formatting as there is no `,` at the end

Good catch! Did not think about that. Made the necessary change in new commit and now there are no issues with black linting.

File: keras/applications/convnext.py [hunk 1]
```diff
@@ -218,7 +218,8 @@ def __init__(self, init_values, projection_dim, **kwargs):
 
     def build(self, input_shape):
         self.gamma = tf.Variable(
```
Comment (qlzh727) [inline keras/applications/convnext.py:220]: Can you replace this with self.add_weight(), which the more standard way to create weights for layer? (it has more logics for handling other aspects.)

**Review by qlzh727 — CHANGES_REQUESTED** [review]
Thanks for the update.

File: keras/applications/convnext.py (hunk 1 above)
Comment (andreped) [reply keras/applications/convnext.py]: Sure. Made a new commit with the suggested change.

File: keras/applications/convnext.py [hunk 2]
```diff
@@ -217,8 +218,11 @@ def __init__(self, init_values, projection_dim, **kwargs):
         self.projection_dim = projection_dim
 
     def build(self, input_shape):
-        self.gamma = tf.Variable(
-            self.init_values * tf.ones((self.projection_dim,))
+        self.gamma = self.add_weight(
+            shape=(self.projection_dim,),
+            dtype=self._compute_dtype_object,
```
Comment (qlzh727) [inline keras/applications/convnext.py:223]: The dtype is not needed since it will be inferred from the initializer.

**Review by qlzh727 — APPROVED** [review]

File: keras/applications/convnext.py (hunk 2 above)
Comment (andreped) [reply keras/applications/convnext.py]: Ok, no worries. Made a new commit removing the dtype.

**Review by qlzh727 — APPROVED** [review]

**Comment by andreped** [comment]
@qlzh727 Seems like the batchnorm float16 unit test failed for the Ubuntu-GPU workflow. Do not see why this PR should be the cause.

Does not look like an assertion error, but rather a CUDA launch issue. I assume you have seen this before. Is this a random error during launch, or is it maybe due to a memory leak when running different tests sequentially on the same GPU?

Does this impact this PR?

**Comment by qlzh727** [comment]
I think it is probably an test infra flaky when running the GPU test. We can ignore it here.

**Review by fchollet — APPROVED** [review]
Yep, the `tf.ones` call will not follow the Keras defaults for MP. Your fix looks good, thanks!
